File: src/main/java/at/hannibal2/skyhanni/config/features/mining/CommissionsBlocksColorConfig.java
package at.hannibal2.skyhanni.config.features.mining;

import at.hannibal2.skyhanni.config.FeatureToggle;
import at.hannibal2.skyhanni.utils.LorenzColor;
import com.google.gson.annotations.Expose;
import io.github.notenoughupdates.moulconfig.annotations.ConfigEditorBoolean;
import io.github.notenoughupdates.moulconfig.annotations.ConfigEditorDropdown;
import io.github.notenoughupdates.moulconfig.annotations.ConfigOption;
import io.github.notenoughupdates.moulconfig.observer.Property;

public class CommissionsBlocksColorConfig {
    @Expose
    @ConfigOption(name = "Enabled", desc = "Change the colour of ores on mining island depending on your active commissions. Grey out irrelevant ores.")
    @ConfigEditorBoolean
    @FeatureToggle
    public boolean enabled = false;

    @Expose
    @ConfigOption(name = "Sneak Toggle", desc = "Quickly disable or enable this feature via sneaking.")
    @ConfigEditorBoolean
    public Property<Boolean> sneakQuickToggle = Property.of(false);

    @Expose
    @ConfigOption(name = "Colour", desc = "Change the highlight colour.")
    @ConfigEditorDropdown
    public Property<LorenzColor> color = Property.of(LorenzColor.GREEN);
}

File: src/main/java/at/hannibal2/skyhanni/config/features/mining/CommissionsBlocksColorConfig.kt
package at.hannibal2.skyhanni.config.features.mining

import at.hannibal2.skyhanni.config.FeatureToggle
import at.hannibal2.skyhanni.utils.LorenzColor
import com.google.gson.annotations.Expose
import io.github.notenoughupdates.moulconfig.annotations.ConfigEditorBoolean
import io.github.notenoughupdates.moulconfig.annotations.ConfigEditorDropdown
import io.github.notenoughupdates.moulconfig.annotations.ConfigOption
import io.github.notenoughupdates.moulconfig.observer.Property

class CommissionsBlocksColorConfig {
    @Expose
    @ConfigOption(
        name = "Enabled",
        desc = "Change the color of ores on mining island depending on your active commissions. Gray out irrelevant ores."
    )
    @ConfigEditorBoolean
    @FeatureToggle
    var enabled: Boolean = false

    @Expose
    @ConfigOption(name = "Sneak Toggle", desc = "Quickly disable or enable this feature via sneaking.")
    @ConfigEditorBoolean
    var sneakQuickToggle: Property<Boolean> = Property.of(false)

    @Expose
    @ConfigOption(name = "Color", desc = "Change the highlight color.")
    @ConfigEditorDropdown
    var color: Property<LorenzColor> = Property.of(LorenzColor.GREEN)
}
